core/src/main/java/hudson/model/Queue.java

```diff
@@ -2303,11 +2309,11 @@ public String getSearchUrl() {
         /** @deprecated Use {@link #doCancelItem} instead. */
         @Deprecated
         @RequirePOST
-        public HttpResponse doCancelQueue() throws IOException, ServletException {
+        public HttpResponse doCancelQueue() {
             if(hasCancelPermission()){
                 Jenkins.get().getQueue().cancel(this);
             }
-            return HttpResponses.forwardToPreviousPage();
+            return HttpResponses.status(HttpServletResponse.SC_NO_CONTENT);
```

Comment:
Is this used anywhere to be called from the UI or something similar? I'm afraid that stop calling forwardToPrevious() changes the behaviour for the cases where this is not expected to behave 100% like an endpoint.

Reply:
It's called in two places in the UI and each places we don't dismiss the redirection. I did the test  canceling a job from the UI.
I just noticed I missed an integration test that I must admit I didn't run locally, I'll fix that :)

Reply:
Fixed